Classes/New Issue/NewIssueTableViewController.swift

```diff
@@ -205,6 +213,10 @@ final class NewIssueTableViewController: UITableViewController, UITextFieldDeleg
         bodyField.inputAccessoryView = actions
     }
 
+    func updateSubmitButtonState() {
+        navigationItem.rightBarButtonItem?.isEnabled = !( titleText == nil || bodyText == nil )
```

Comment:
I think we should split this up a bit, with the goal of making it easier to read (and again, not requiring a body).

What do you think about the following?

```swift
let hasTitleText = titleText != nil
let isTitleTextEmpty = titleText?.isEmpty == true // This would be additional. We could even check for whitespace only in the future if we want to.
navigationItem.rightBarButtonItem?.isEnabled = hasTitleText && isTitleTextEmpty == false
```

Reply:
Our titleText variable, higher up in the code, checks inside the field for empty and whitespace. 

```suggestion
private var titleText: String? {
     guard let raw = titleField.text?.trimmingCharacters(in: .whitespacesAndNewlines) else { return nil }
     if raw.isEmpty { return nil }
     return raw
}
```

We could just go back to this original action code for the titleField.

```suggestion
@IBAction func titleFieldEditingChanged(_ sender: Any) {
      navigationItem.rightBarButtonItem?.isEnabled = titleText != nil
 }
```

Reply:
Higher up our titleText variable is checking for empty and whitespace.

```swift
private var titleText: String? {
        guard let raw = titleField.text?.trimmingCharacters(in: .whitespacesAndNewlines) else { return nil }
        if raw.isEmpty { return nil }
        return raw
}
```

I think the best option would be to revert to the original. If we are doing away with the UITextViewDelegate and not checking bodyField to turn on the Submit button we can get rid of that function completely. 

```swift
@IBAction func titleFieldEditingChanged(_ sender: Any) {
        navigationItem.rightBarButtonItem?.isEnabled = titleText != nil
}
```

Reply:
Ah, I missed that. Go ahead.